feature-libs/quote/components/items/quote-items.component.ts

```diff
@@ -42,29 +49,66 @@ export class QuoteItemsComponent {
   protected quoteItemsComponentService = inject(QuoteItemsComponentService);
   protected userIdService = inject(UserIdService);
 
-  quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
-  quoteCartReadOnly$: Observable<Cart> = this.userIdService.takeUserId().pipe(
-    switchMap((userId) => zip(this.quoteDetails$, of(userId))),
-    tap(([quote, userId]) => {
-      if (quote.cartId !== undefined) {
-        this.multiCartFacade.loadCart({
-          userId,
-          cartId: quote.cartId,
-          extraData: { active: false },
-        });
-      }
-    }),
-    switchMap(([quote, _userId]) => {
-      return quote.cartId !== undefined
-        ? this.multiCartFacade.getCart(quote.cartId)
-        : NEVER;
-    })
-  );
+  quoteItemsData$: Observable<QuoteItemsData> = this.userIdService
```

Comment:
[MINOR]
It's quite a long stream. I only guess that some customers might want to customize parts of it.
Perhaps it's worth moving it (or parts of it) to separate function(s) in a component service?